pubpub/platform
# Add detailed component configuration

## Issue(s) Resolved
#551 
## High-level Explanation of PR
- Modifies FormElement to once again use the component rather than the schema to choose what to render
- Adds configuration forms for all existing component types
- Renders labels etc. based on that configuration

Let me know if you think the structure should be improved to make it easier to add new elements. Adding a new component requires a bit more boilerplate than I would have liked, but I don't think it's enough overhead that we need to do anything about it yet.

## Test Plan
- Visit the seeded form in the croccroc community and configure the various fields
- Construct the form fill url by getting an existing pubId, then make sure the fields appear and render alright, and that validation, saving, and submission work. 
http://localhost:3000/c/croccroc/public/forms/review/fill?pubId={pubId}
- Make sure the pub update and pub create forms aren't broken 

## Screenshots (if applicable)
